src/adagrams.js

```diff
@@ -1,15 +1,143 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const lettersArr = Object.entries(LETTER_POOL)
+    .map(([key, value]) => {
+      return key.repeat(value);
+    })
+    .join('')
+    .split('');
+
+  const indices = [];
+  const letters = [];
+  while (letters.length < 10) {
+    const randomIndex = Math.floor(Math.random() * lettersArr.length);
+    if (!indices.includes(randomIndex)) {
```

Comment:
I like this approach as it attempts to not mutate `lettersArr` but I think in this case you might actually want to mutate the original array because the randomness here is based on the unchanged length of `lettersArr`.